tests/foreman/ui_airgun/test_repository.py

```diff
@@ -482,3 +493,279 @@ def test_positive_resynchronize_puppet_repo(session, module_prod):
+                'repo_content.gpg_key': gpg_key.name,
+                'repo_content.download_policy': DOWNLOAD_POLICIES['on_demand']
+            }
+        )
+        assert session.repository.search(module_prod.name, repo_name)[0]['Name'] == repo_name
+        repo_values = session.repository.read(module_prod.name, repo_name)
+        assert repo_values['repo_content']['upstream_url'] == FAKE_1_YUM_REPO
+        assert repo_values['repo_content']['metadata_type'] == checksum_type
+        assert repo_values['repo_content']['gpg_key'] == gpg_key.name
+        assert repo_values['repo_content']['download_policy'] == DOWNLOAD_POLICIES['on_demand']
+        session.repository.update(
+            module_prod.name,
+            repo_name,
+            {
+                'name': new_repo_name,
+                'repo_content.upstream_url': FAKE_2_YUM_REPO,
+                'repo_content.metadata_type': new_checksum_type,
+                'repo_content.gpg_key': new_gpg_key.name,
+                'repo_content.download_policy': DOWNLOAD_POLICIES['immediate'],
+            }
+        )
+        assert not session.repository.search(module_prod.name, repo_name)
+        repo_values = session.repository.read(module_prod.name, new_repo_name)
+        assert repo_values['name'] == new_repo_name
+        assert repo_values['repo_content']['upstream_url'] == FAKE_2_YUM_REPO
+        assert repo_values['repo_content']['metadata_type'] == new_checksum_type
+        assert repo_values['repo_content']['gpg_key'] == new_gpg_key.name
+        assert repo_values['repo_content']['download_policy'] == DOWNLOAD_POLICIES['immediate']
+        session.repository.delete(module_prod.name, new_repo_name)
+        values = session.repository.search(module_prod.name, new_repo_name)
+        # a task is created for delete operation and we are not redirected to that task page.
+        # this operation can take some time.
+        for _ in range(10):
+            values = session.repository.search(module_prod.name, new_repo_name)
+            if not values:
+                break
+            time.sleep(2)
+        assert not values
+
+
+@tier2
+@upgrade
+def test_positive_upstream_with_credentials(session, module_prod):
+    """Create repository with upstream username and password update them and then clear them.
+
+    :id: 141a95f3-79c4-48f8-9c95-e4b128045cb3
+
+    :expectedresults:
+
+        1. The custom repository is created with upstream credentials.
+        2. The custom repository upstream credentials are updated.
+        3. The credentials are cleared.
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+
+    :BZ: 1433481
+    """
+    repo_name = gen_string('alpha')
+    upstream_username = gen_string('alpha')
+    upstream_password = gen_string('alphanumeric')
+    new_upstream_username = gen_string('alpha')
+    new_upstream_password = gen_string('alphanumeric')
+    hidden_password = '*' * 8
```

Comment:
I am really think it is not good idea to verify passwords in that test case as it is not actual validation

Reply:
It won't verify password, but will make sure password is hidden and not shown as a plain text. I'd keep such assertion

Reply:
at least update doesn't make much sense?

Reply:
it will ensure password won't expose when updating. Yeah, real value is pretty low, but i can't say it's completely useless and it's definitely not time/resource consuming and is a part of name assertion anyways, so why not?